docs/extensions/extension-migration.md

```diff
@@ -0,0 +1,22 @@
+# Lens v4 to v5 Extension Migration Notes
+
+*  Lens v5 inspects the version of the extension to ensure it is compatible. 
+The `package.json` for your extension must have an `"engines"` field specifying the lens version that your extension is targeted for, e.g:
```

Comment:
I would add that we support all the range semantics that https://www.npmjs.com/package/semver supports.